keylime/config.py

```diff
@@ -302,6 +306,94 @@ def yaml_to_dict(
     return None
 
 
+def str_to_version(v_str: str) -> Union[Tuple[int, int], None]:
+    """
+    Validates the string format and converts the provided string to a tuple of
+    ints which can be sorted and compared.
+
+    :returns: Tuple with version number parts converted to int. In case of
+    invalid version string, returns None
+    """
+
+    # Strip to remove eventual quotes and spaces
+    v_str = v_str.strip('" ')
+
+    m = re.match(r"^(\d+)\.(\d+)$", v_str)
+
+    if not m:
+        return None
+
+    return tuple(int(x) for x in m.group(1, 2))
+
+
+def check_version(component: str, logger: Optional[logging.Logger] = None) -> bool:
+    """
+    Check component current configuration file version and return a boolean
+    indicating whether an upgrade is available
+    """
+
+    if not os.path.exists(TEMPLATES_DIR):
+        # If there are no templates available
+        return False
+
+    if not os.path.isdir(TEMPLATES_DIR):
+        if logger:
+            logger.warning("The path %s is not a directory", TEMPLATES_DIR)
+        return False
+
+    dirs = os.listdir(TEMPLATES_DIR)
+
+    if not dirs:
+        if logger:
+            logger.warning(
+                "The path %s does not contain version directories for config upgrade templates", TEMPLATES_DIR
+            )
+        return False
+
+    # Sort in reverse order to get first the latest version available
+    versions = sorted((x for x in set(map(str_to_version, dirs)) if x is not None), reverse=True)
+
+    if not versions:
+        if logger:
+            logger.warning("The path %s does not contain valid config version upgrade directories", TEMPLATES_DIR)
+        return False
+
+    config_version = get(component, "version")
+
+    if not config_version:
+        config_version = "1.0"
```

Comment:
Hum, I think now it can. Previously, the code was mixing `dict` and `RawConfigParser` which made `get()` to behave differently.  Now, the code uses only `RawConfigParser` which makes `get()` to be consistent.

Reply:
Done